iron-form.html

```diff
@@ -11,503 +11,459 @@
+        if (!this.request) {
+          this.request = document.createElement('iron-ajax');
+          this.request.addEventListener('response', this._handleFormResponse.bind(this));
+          this.request.addEventListener('error', this._handleFormError.bind(this));
+        }
 
-      // Save the original value of this input.
-      this._customElementsInitialValues.push(
-          this._usesCheckedInsteadOfValue(element) ? element.checked : element.value);
-    },
+        // Native forms can also index elements magically by their name (can't make
+        // this up if I tried) so we need to get the correct attributes, not the
+        // elements with those names.
+        this.request.url = this._form.getAttribute('action');
+        this.request.method = this._form.getAttribute('method') || 'GET';
+        this.request.contentType = this._form.getAttribute('enctype');
+        this.request.withCredentials = this.withCredentials;
+        this.request.headers = this.headers;
+
+        if (this._form.method.toUpperCase() === 'POST') {
+          this.request.body = json;
+        } else {
+          this.request.params = json;
+        }
 
-    _unregisterElement: function(e) {
-      var target = e.detail.target;
-      if (target) {
-        var index = this._customElements.indexOf(target);
-        if (index > -1) {
-          this._customElements.splice(index, 1);
-          this._customElementsInitialValues.splice(index, 1);
+        // Allow for a presubmit hook
+        var event = this.fire('iron-form-presubmit', {}, {cancelable: true});
+        if(!event.defaultPrevented) {
+          this.request.generateRequest();
+          this.fire('iron-form-submit', json);
         }
-      }
-    },
-
-    /**
-     * Validates all the required elements (custom and native) in the form.
-     * @return {boolean} True if all the elements are valid.
-     */
-    validate: function() {
-      var valid = true;
-
-      // Validate all the custom elements.
-      var validatable;
-      for (var el, i = 0; el = this._customElements[i], i < this._customElements.length; i++) {
-        if (!this._isChildOfRegisteredParent(el, false) && !el.disabled) {
-          validatable = /** @type {{validate: (function() : boolean)}} */ (el);
-          // Some elements may not have correctly defined a validate method.
-          if (validatable.validate)
-            valid = !!validatable.validate() && valid;
+      },
+
+      _getValidatableElements: function() {
+        var nodes = Polymer.dom(this._form).children;
+        var validatable = [];
+
+        for (var i = 0; i < nodes.length; i++) {
+          // This element might contain other potentially validatable elements
+          // which we need to look at. For example, this could be a <p> node
+          // containing a bunch of inputs.
+          var elements = Array.prototype.slice.call(Polymer.dom(nodes[i]).querySelectorAll('*')) || [];
```

Comment:
Instead of getting the children and for each child do querySelectorAll, how about just do `Polymer.dom(this._form).querySelectorAll('*')`?